Makefile.toml

```
@@ -11,267 +10,483 @@ skip_core_tasks = true
 skip_git_env_info = true
 skip_rust_env_info = true
 
-
 [tasks.default]
-description = "List tasks"
+description = "List available tasks"
+category = "Default"
 script = "cargo make --quiet --list-all-steps --hide-uninteresting"
 
-
-# Build
+# =================================================================================================
+# Development Flows - Quick iteration for local development
+# =================================================================================================
+[tasks.dev]
+description = "Development flow with all features"
+category = "Development"
+run_task = { name = ["format", "build", "clippy"] }
+
+[tasks.dev-default]
+description = "Development flow with default features"
+category = "Development"
+run_task = { name = ["format", "build-default", "clippy"] }
+
+[tasks.dev-preview]
+description = "Development flow with preview features"
+category = "Development"
+run_task = { name = ["format", "build-preview", "clippy"] }
+
+[tasks.dev-experimental]
+description = "Development flow with experimental features"
+category = "Development"
+run_task = { name = ["format", "build-experimental", "clippy"] }
+
+# =================================================================================================
+# Quality Assurance - Full checks before committing
+# =================================================================================================
+[tasks.qa]
+description = "Quality assurance checks"
+category = "Quality"
+run_task = { name = ["check-format", "clippy", "check-features", "security-flow", "test", "example-flow"] }
+
+# =================================================================================================
+# CI Flows - Comprehensive testing for continuous integration
+# =================================================================================================
+[tasks.ci]
```

Let's call this `cargo make all` to capture the intent of running all the tasks rather than claiming this is an exact match to CI. CI may change in the future and is unlikely to run `cargo make ci` directly because we will want to parallelize tasks on different jobs/runners.
```suggestion
[tasks.all]
```